### ISSUE TITLE
Remote state resource vs command line

### ISSUE THREAD

**Issue by spanktar** [hashicorp/terraform#7759]
### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.

0.6.15 (have blockers in later releases)

### Affected Resource(s)
Please list the resources as a list, for example:
- terraform_remote_state

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Actual Behavior
When configuring Terraform to use terraform remote state, if you add a resource declaration like `terraform_remote_state { foo }`, you would expect terraform to use that for storing the state file when running `terraform plan` or similar. Instead, you have to initialize it using the command line like so `terraform remote config etc. etc.  The problem I have is, once you've set it via command line, the resource declaration isn't necessary. I've been using a "run once" shell script to initialize a terraform build that does `terraform remote config...` without a terraform_remote_state{} block for months.

### Expected Behavior
What one would expect is that when you define `terraform_remote_state {}` you shouldn't need to run another shell script.  Just use the declared remote state resource.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Define a terraform_remote_state block
2. run `terraform plan`
3. Note how the remote state is not initialized or used.
4. remove terraform_remote_state
5. run `terraform remote config etc. etc.`
6. note how remote state is initialized and used without a resource definition
7. run `terraform apply` over and over without ever having to re-initialize.

Weird, right?

**Comment by jamielennox1** [comment]
The terraform_remote_state resource is read only access to other remote states.

For example; you have a team who deals with the networking layer of the infrastructure. They'll make outputs of things like subnet_ids, peering_ids.

This will look like so:

`resource "terraform_remote_state" "network" {
  backend = "s3"

  config {
    bucket = "s3bucketname"
    key    = "/network/terraform.tfstate"
  }
}`

Another team dealing with instances such as NGINX; they can access the subnet_ids from the infrastructure teams in a read only fashion using terraform_remote_state.
The state you now have for your instances has now be pushed to the remote state.

In your resources you can now use the subnet_ids given by the infrastructure team like so:

`resource "aws_instance" "esb" {
  count                = "${var.instance_count}"
  availability_zone    = "${element(split(",", terraform_remote_state.network.output.azs), count.index)}"
  subnet_id            = "${element(split(",", terraform_remote_state.network.output.private_subnet_ids), count.index)}"
  instance_type        = "${var.instance_type}"
  ami                  = "${var.ami}"
}`

If the team who are in charge of the instances want to make changes to their own resources then they will have to initialise this using the "terraform remote config ..."

in summary: 
`terraform remote config` is for your own state
`terraform_remote_state` is to get read only access to someone else's state

**Comment by spanktar** [comment]
OK, your explanation is clear, but the thinking behind it doesn't sit well with me.

Maybe I guess what I'm asking for in light of this explanation is a `terraform_remote_config` resource.  I just don't think you should need to have a separate shell script (or wrapper or something — there are all kinds of home-rolled solutions out there) to initialize the remote state. It just add another thing in .gitignore and another setup step for every team member to do, and if it doesn't go right, the remote state gets blanked, orphaning all of the resources (it's happened to me a few times).

I guess for now, I'll just keep using my `remote.sh` script.

Regardless, thanks for explaining this.
